lib/block_view_lockups.go

```diff
@@ -0,0 +1,2179 @@
+			return errors.Wrapf(err, "_disconnectCoinLockup: Problem unSpending balance of %v "+
+				"for the transactor", operationData.BalanceAmountNanos)
+		}
+	} else {
+		// Revert the transactor's DAO coin balance.
+		bav._setBalanceEntryMappings(operationData.PrevTransactorBalanceEntry, true)
+
+		// Fetch the profile entry associated with the lockup.
+		profileEntry := bav.GetProfileEntryForPKID(operationData.PrevLockedBalanceEntry.ProfilePKID)
+		if profileEntry == nil || profileEntry.isDeleted {
+			return fmt.Errorf("_disconnectCoinLockup: Trying to revert coin entry " +
+				"update but found nil profile entry; this shouldn't be possible")
+		}
+
+		// Ensure the PrevCoinEntry is not nil. This shouldn't be possible.
+		if operationData.PrevCoinEntry == nil {
+			return fmt.Errorf("_disconnectCoinLockup: Trying to revert coin entry " +
+				"update but found nil prev coin entry; this shouldn't be possible")
+		}
+
+		// Revert the coin entry.
+		profileEntry.DAOCoinEntry = *operationData.PrevCoinEntry
+		bav._setProfileEntryMappings(profileEntry)
+	}
+
+	// By here we only need to disconnect the basic transfer associated with the transaction.
+	basicTransferOps := utxoOpsForTxn[:operationIndex]
+	err = bav._disconnectBasicTransfer(currentTxn, txnHash, basicTransferOps, blockHeight)
+	if err != nil {
+		return errors.Wrapf(err, "_disconnectCoinLockup")
+	}
+	return nil
+}
+
+//
+// UpdateCoinLockupParams Transaction Logic
+//
+
+func (bav *UtxoView) _connectUpdateCoinLockupParams(
+	txn *MsgDeSoTxn,
+	txHash *BlockHash,
+	blockHeight uint32,
+	verifySignatures bool,
+) (_totalInput uint64,
+	_totalOutput uint64,
+	_utxoOps []*UtxoOperation,
+	_err error) {
+	var utxoOpsForTxn []*UtxoOperation
+
+	// Validate the starting block height.
+	if blockHeight < bav.Params.ForkHeights.ProofOfStake1StateSetupBlockHeight ||
+		blockHeight < bav.Params.ForkHeights.BalanceModelBlockHeight {
+		return 0, 0, nil,
+			errors.Wrapf(RuleErrorLockupTxnBeforeBlockHeight, "_connectUpdateCoinLockupParams")
+	}
+
+	// Validate the txn TxnType.
+	if txn.TxnMeta.GetTxnType() != TxnTypeUpdateCoinLockupParams {
+		return 0, 0, nil, fmt.Errorf("_connectUpdateCoinLockupParams: "+
+			"called with bad TxnType %s", txn.TxnMeta.GetTxnType().String())
+	}
+
+	// Try connecting the basic transfer without considering transaction metadata.
+	totalInput, totalOutput, utxoOpsForBasicTransfer, err :=
+		bav._connectBasicTransfer(txn, txHash, blockHeight, verifySignatures)
+	if err != nil {
+		return 0, 0, nil, errors.Wrapf(err, "_connectUpdateCoinLockupParams")
+	}
+	utxoOpsForTxn = append(utxoOpsForTxn, utxoOpsForBasicTransfer...)
+
+	// Grab the txn metadata.
+	txMeta := txn.TxnMeta.(*UpdateCoinLockupParamsMetadata)
+
+	// Get the profilePKID from the transactor public key.
+	var profilePKID *PKID
+	_, updaterIsParamUpdater := GetParamUpdaterPublicKeys(blockHeight, bav.Params)[MakePkMapKey(txn.PublicKey)]
+	if updaterIsParamUpdater {
+		profilePKID = ZeroPKID.NewPKID()
```

Comment:
so any param updater cannot have a coin that has any lock up params? This seems okay, but we should make this explicit here with a comment.